addons/mail/static/tests/message/message.test.js

```diff
@@ -1840,5 +1841,6 @@ test("Delete starred message decrements starred counter once", async () => {
     await click(":nth-child(1 of .o-mail-Message) [title='Expand']");
     await click(".o-mail-Message-moreMenu [title='Delete']");
     await click("button", { text: "Confirm" });
+    await delay(50);
```

Comment:
Delays are not a robust solution for testing. A fixed delay of 50 ms may work in some cases, but it does not guarantee reliability due to several factors:

- High CPU load on the runbot.
- Variability across different machines.
- Different scenarios with the task queue (timeouts, promises, postMessages, etc.).

As a result, this test may fail at some point, or it might pass due to inaccurate timing. Instead of using arbitrary delays, we should wait for the actual event to occur. For example, if we are waiting for a message deletion, we could wait for the event to be processed using steps?

Reply:
Tried by setting up a step for `/mail/message/update_content` (message deletion) or waiting for the message to be deleted on the frontend but the effect seems to happen later as the test still passes. Could maybe try patching some `store` methods but the main problem is that the if the counter goes like: 3 -> 2 -> 1 the test will pass as it will find `"Starred2"` at some point.

Reply:
Then we have to understand what is making the counter decrement and wait for that particular event? (even if we could argue that one day an even later event could also decrement the counter but that's also the case for the delay approach).

Reply:
The issue is that this test is based on the side bar only and that it's not enough as you found out. If we accessed the starred inbox then did the same test and waited for the message to be deleted from the thread before making the assertion, there would be a good chance that the processing of the message deletion is done? 
- Message is deleted => we know that the event was received and processed
- Change repercuted in the DOM => we know a bunch of time passed, owl waits 1 micro task tick, then the new DOM have to be computed and patched, that's a lot of time.

Reply:
Still not enough:
```js
test("Delete starred message decrements starred counter once", async () => {
    const pyEnv = await startServer();
    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
    pyEnv["mail.message"].create([
        {
            author_id: serverState.partnerId,
            body: "delete me",
            message_type: "comment",
            model: "discuss.channel",
            res_id: channelId,
            starred_partner_ids: [serverState.partnerId],
        },
        {
            author_id: serverState.partnerId,
            body: "Hello World!",
            message_type: "comment",
            model: "discuss.channel",
            res_id: channelId,
            starred_partner_ids: [serverState.partnerId],
        },
        {
            author_id: serverState.partnerId,
            body: "test",
            message_type: "comment",
            model: "discuss.channel",
            res_id: channelId,
            starred_partner_ids: [serverState.partnerId],
        },
    ]);
    await start();
    await openDiscuss(channelId);
    await contains("button", { count: 1, text: "Starred3" });
    await click(":nth-child(1 of .o-mail-Message) [title='Expand']");
    await click(".o-mail-Message-moreMenu [title='Delete']");
    await click("button", { text: "Confirm" });
    await click("button", { text: "Starred" }); // diff
    await contains(".o-mail-Message", { count: 0, text: "delete me" }); // diff
    await contains("button", { count: 1, text: "Starred2" });
});
```

![image](https://github.com/user-attachments/assets/23e5d16c-6017-448e-aaed-06a04136cf7a)

(test passes)